### PR TITLE
Here's a few suggested changes

### PR DESCRIPTION
**Pull request by strawstack**
Not sure if everything will completely work after these changes, but I think it should fix some bugs

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -89,7 +89,7 @@ fn run_program(ints: &mut Vec<u16>) {
             4 => {
                 // set <a> to 1 if <b> is equal to <c>; set it to 0 otherwise
                 if b == c {
-                    write(&mut memory, &mut registers, (cursor + 1) as u16, 1);
+                    write(&mut memory, &mut registers, memory[(cursor + 1) as usize], 1); // COMMENT: This change I think should be made to all write calls; I think you want the memory at the cursor rather than the cursor itself
```
Comment (AngelOnFira) [inline src/main.rs]: @strawstack hmm but if I change this, wouldn't I also have to change 5?

Comment (AngelOnFira) [reply src/main.rs]: oh never mind, didn't read the comment :P I'll check this out :100:

File: src/main.rs [hunk 2]
```diff
@@ -179,7 +179,7 @@ fn run_program(ints: &mut Vec<u16>) {
             17 => {
                 // write the address of the next instruction to the stack and
                 // jump to <a>
-                stack.push(b);
+                stack.push(cursor + 1); // COMMENT: cursor + 1 is the next instruction
```
Comment (AngelOnFira) [inline src/main.rs]: truuuuuuuu ty

**Comment by strawstack** [comment]
I'll close this because I have new better changes that make everything work

**Pull request closed without merging**